packages/keycloak-user-management/src/components/Credentials/index.tsx

```diff
@@ -0,0 +1,189 @@
+        description: '',
+      });
+    });
+};
+
+const UserCredentials: React.FC<PropsTypes> = (props: PropsTypes) => {
+  const userId = props.match.params.userId;
+  const isEditMode = !!userId;
+  const layout = {
+    labelCol: { span: 4 },
+    wrapperCol: { span: 16 },
+  };
+
+  //   if (isLoading) {
+  //     return <Ripple />;
+  //   }
+
+  return (
+    <Col span={12}>
+      <Card title={`${isEditMode ? 'Edit User' : 'Create New User'}`} bordered={false}>
+        <HeaderBreadCrumb userId={userId} />
+        <div className="form-container">
+          <Form
+            {...layout}
+            onFinish={(values: UserCredentialsFormFields) => submitForm(values, props)}
+          >
+            <Form.Item
+              name="password"
+              label="Password"
+              rules={[
+                {
+                  required: true,
+                  message: 'Please input your password!',
+                },
+              ]}
+              hasFeedback
+            >
+              <Input.Password />
+            </Form.Item>
+
+            <Form.Item
+              name="confirm"
+              label="Confirm Password"
+              dependencies={['password']}
+              hasFeedback
+              rules={[
+                {
+                  required: true,
+                  message: 'Please confirm your password!',
+                },
+                ({ getFieldValue }) => ({
+                  validator(rule, value) {
+                    if (!value || getFieldValue('password') === value) {
+                      return Promise.resolve();
+                    }
+                    return Promise.reject('The two passwords that you entered do not match!');
+                  },
+                }),
+              ]}
+            >
+              <Input.Password />
+            </Form.Item>
+            <Form.Item name="temporary" label="Temporary" valuePropName="checked">
+              <Switch />
+            </Form.Item>
+            <Form.Item>
+              <Row justify="start">
+                <Col span={4}>
+                  <Button htmlType="submit" className="reset-password">
+                    Reset Password
+                  </Button>
+                </Col>
+              </Row>
+            </Form.Item>
+          </Form>
+        </div>
+      </Card>
+    </Col>
+  );
+};
+
+UserCredentials.defaultProps = defaultProps;
+
+export { UserCredentials };
+
+/** Interface for connected state to props */
+interface DispatchedProps {
+  keycloakUser: KeycloakUser | null;
+  accessToken: string;
+}
+
+// connect to store
+const mapStateToProps = (state: Partial<Store>, ownProps: PropsTypes): DispatchedProps => {
+  const userId = ownProps.match.params.userId;
+  const keycloakUsersSelector = makeKeycloakUsersSelector();
+  const keycloakUsers = keycloakUsersSelector(state, { id: [userId] });
+  const keycloakUser = keycloakUsers.length === 1 ? keycloakUsers[0] : null;
```

Comment:
- What happens if keycloakUsers.length > 1?

Reply:
When editing a user only that single user is dispatched to store so there will only be one user in the users list.

Reply:
Maybe we should then change this code to be keycloakUsers.length >= 1?  Just to future proof it?